Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -80,65 +78,88 @@ public final class ActorQueue {
 
     // MARK: Public
 
+    /// Sets the actor context within which each `async` and `await`ed task will execute.
+    /// It is recommended that this method be called in the adopted actor’s `init` method.
+    /// **Must be called prior to enqueuing any work on the receiver.**
+    ///
+    /// - Parameter actor: The actor on which the queue's task will execute. This parameter is not retained by the receiver.
+    /// - Warning: Calling this method more than once will result in an assertion failure.
+    public func adoptExecutionContext(of actor: ActorType) {
+        assert(executionContext == nil) // Setting multiple executionContexts on the same queue is API abuse.
+        executionContext = actor
+    }
+
     /// Schedules an asynchronous task for execution and immediately returns.
     /// The scheduled task will not execute until all prior tasks have completed or suspended.
-    /// - Parameter task: The task to enqueue.
-    public func async(_ task: @escaping @Sendable () async -> Void) {
-        taskStreamContinuation.yield(task)
+    /// - Parameter task: The task to enqueue. The task's parameter is a reference to the actor whose execution context has been adopted.
+    public func async(_ task: @escaping @Sendable (isolated ActorType) async -> Void) {
+        // Crashing here means that this queue is being sent tasks either before an execution context has been set, or
```

Comment:
Nice helpful comment for a consumer of the library 👍

Reply:
🎉 yeah I figure if I'm going to be using `unowned` and force unwraps I might as well guide the consumer when the API inevitably is abused.